config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/ServiceEntityBuilder.java

```diff
@@ -67,24 +71,38 @@ private Entity buildServiceEntity(Bundle bundle, String servicePath, Service ser
         serviceDetailElement.appendChild(buildServiceMappings(service, document));
 
         if (service.getProperties() != null) {
-            buildAndAppendPropertiesElement(service.getProperties()
-                            .entrySet()
-                            .stream()
-                            .collect(Collectors
-                                    .toMap(p -> "property." + p.getKey(), p -> p.getKey().startsWith(PREFIX_ENV) ? "SERVICE_PROPERTY_" + p.getKey() : p.getValue())),
+            Map<String, Object> properties = service.getProperties()
+                    .entrySet()
+                    .stream()
+                    .collect(Collectors
+                            .toMap(p -> "property." + p.getKey(), p -> p.getKey().startsWith(PREFIX_ENV) ? "SERVICE_PROPERTY_" + p.getKey() : p.getValue()));
+
+            if(isSoapService) {
```

Comment:
I think your properties have to stay out of this condition service.getProperties != null
If you have a soap service, that does not have any other properties, you will end up with a bundle with the WSDL and everything, but without the isSoap = true and the soapVersion because the getProperties was null.

Reply:
yup.. you are right.. I will segregate this.. thanks